src/main/scala/it/pagopa/interop/tenantprocess/api/impl/TenantApiServiceImpl.scala

```diff
@@ -669,6 +669,7 @@ final case class TenantApiServiceImpl(
     agreements <- agreementManagementService.getAgreements(producerId, consumerId, agreementStates)
     eServices  <- Future.traverse(agreements.map(_.eserviceId))(catalogManagementService.getEServiceById)
     attributeIds = eServices
+      .flatMap(_.descriptors.sortBy(_.version).lastOption.toList)
```

Comment:
Instead of the last descriptor, I think it should take the descriptors corresponding to the `descriptorId` in `agreements`

Reply:
Hmm, let me try

Reply:
Something like this might fit?
```diff
    agreements <- agreementManagementService.getAgreements(producerId, consumerId, agreementStates)
+   descriptorIds = agreements.map(_.descriptorId)
    eServices <- Future.traverse(agreements.map(_.eserviceId))(catalogManagementService.getEServiceById)
    attributeIds = eServices
+     .flatMap(_.descriptors.filter(d => descriptorIds.contains(d.id)))
```

Reply:
In this way we search for that specific attribute in any descriptor for whom this consumer/producer have an agreement

Reply:
it works for me 🚀